### PR TITLE
(PCP-92) Add logrotate configuration

### PR DESCRIPTION
**Pull request by geoffnichols**
This commit adds a logrotate config for the pxp-agent.

### CODE REVIEW

File: ext/pxp-agent.logrotate [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/var/log/puppetlabs/pxp-agent/pxp-agent.log {
```
Comment (haus) [inline ext/pxp-agent.logrotate]: Is this the only log generated by pxp-agent?

Comment (stahnma) [inline ext/pxp-agent.logrotate]: Seems like doing *.log here would be more future proof.

**Comment by haus** [comment]
Who owns the pxp-agent directory? root?

**Comment by haus** [comment]
Does logrotate compress by default? If not, that's probably worth turning on.

**Comment by mhaskel** [comment]
Don't you have to specify one of time or size?

File: ext/pxp-agent.logrotate [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/var/log/puppetlabs/pxp-agent/*.log {
+    daily
+    missingok
+    rotate 30
+    compress
```
Comment (geoffnichols) [inline ext/pxp-agent.logrotate]: It looks like logrotate compresses (with gzip) by default, but I have added that explicitly here.

File: ext/pxp-agent.logrotate [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/var/log/puppetlabs/pxp-agent/*.log {
+    daily
```
Comment (geoffnichols) [inline ext/pxp-agent.logrotate]: I'm not sure if daily or weekly will make the most sense here.

File: ext/pxp-agent.logrotate [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    missingok
+    rotate 30
```
Comment (geoffnichols) [inline ext/pxp-agent.logrotate]: Is this overkill? Maybe 10 would be sufficient?

File: ext/pxp-agent.logrotate (hunk 3 above)
Comment (shrug) [inline ext/pxp-agent.logrotate]: Probably depends on what kind of info is in the log? Daily & keep 30 works for me for now? Or weekly, keep 4-ish?

**Comment by geoffnichols** [comment]
@haus `/var/log/puppetlabs/pxp-agent/` is currently owned by `root:root`.

**Comment by geoffnichols** [comment]
Relates to https://github.com/puppetlabs/pxp-agent/pull/148.

**Comment by mhaskel** [comment]
:+1:

**Pull request merged**